src/lib/context.py

```diff
@@ -78,6 +78,7 @@ def __init__(
         self.maximum_metric_data_points_per_minute = int(os.environ.get("MAXIMUM_METRIC_DATA_POINTS_PER_MINUTE", "100000"))
         self.metric_ingest_batch_size = int(os.environ.get("METRIC_INGEST_BATCH_SIZE", "1000"))
         self.require_valid_certificate = os.environ.get("REQUIRE_VALID_CERTIFICATE", "True") in ["True", "T", "true"]
+        self.use_x_goog_user_project_header = {}
```

Comment:
There is no point in sending this header to owner project, we can optimize a bit by initializing this Dict with `{project_id_owner: False}`